recipes/libnvimgcodec/meta.yaml

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+{% set version = '0.3.0' %}
+{% set somajor = version.split('.')[0] %}
+{% set cuda_major = environ.get("cuda_compiler_version", "11.8").split(".")[0]|int %}
+
+package:
+  name: libnvimgcodec-split
+  version: {{ version }}
+
+source:
+  url: https://github.com/NVIDIA/nvImageCodec/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 4bba949d6cf4e88fcd6b2d86bff960c9ed68eb25b6c4dde3b7772615480ab9fb
+  patches:
+    - patches/0001-BUG-Add-missing-headers-from-standard-library.patch
+    - patches/0002-BLD-pybind11-dlpack-only-needed-when-building-for-py.patch
+    - patches/0003-DOC-Add-Fatal-Error-if-extensions-are-built-without-.patch
+    - patches/0004-BLD-Let-end-user-set-libdir-using-CMake-variables.patch
+    - patches/0005-BLD-Make-moving-libraries-to-python-directory-option.patch
+    - patches/0006-BLD-Use-consistent-install-directory-for-cmake-confi.patch
+    - patches/0007-BLD-Let-python-versions-be-user-selectable.patch
+    - patches/0008-BLD-Decouple-nvimgcodec-and-dynlink_-targets.patch
+    - patches/0009-BLD-Link-system-nvimgcodec-to-python-module.patch
+    - patches/0010-BLD-Find-libclang-for-conda-forge.patch
+
+build:
+  number: 0
+  skip: true  # [not (linux or win)]
+  skip: true  # [cuda_compiler_version in (None, 'None')]
+  # Debug skips below
+  # skip: true  # [py != 312]
+  # skip: true  # [cuda_compiler_version != '12.0']
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cuda') }}
+    - {{ compiler('cxx') }}
+    - {{ stdlib('c') }}
+    - cmake >=3.18
+    - ninja
+    - pkg-config
+    - python-clang
+  host:
+    - cuda-version {{ cuda_compiler_version }}
+    - libboost-headers
+    - libjpeg-turbo
+{% if cuda_major == 12 %}
+    - libnvjpeg-dev
```

Comment:
Should dependencies linked with dlopen be made optional or converted to standard dynamic links?